core/dbt/include/global_project/macros/utils/array_concat.sql

```diff
@@ -0,0 +1,7 @@
+{% macro array_concat(array_1, array_2) -%}
+  {{ return(adapter.dispatch('array_concat', 'dbt')(array_1, array_2)) }}
+{%- endmacro %}
+
+{% macro default__array_concat(array_1, array_2) -%}
+    array_cat({{ array_1 }}, {{ array_2 }})
```

Comment:
if `array_concat` is more common should we use that as the default instead of postgres syntax?

Reply:
Between Postgres, Snowflake, BigQuery, and Redshift, it's [six of one, half a dozen of the other](https://en.wikipedia.org/wiki/Six_of_One).
- [`array_cat`](https://www.postgresql.org/docs/9.1/functions-array.html#:~:text=Table%209%2D42.%20Array%20Functions) - Postgres
- [`array_cat`](https://docs.snowflake.com/en/sql-reference/functions/array_cat.html) - Snowflake
- [`array_concat`](https://docs.aws.amazon.com/redshift/latest/dg/r_array_concat.html) - BigQuery
- [`array_concat`](https://cloud.google.com/bigquery/docs/reference/standard-sql/array_functions#array_concat) - Redshift

Spark/Databricks refuses to break the tie by using a unique function name for array concatenation:
- [`concat`](https://spark.apache.org/docs/latest/api/sql/index.html#concat) - Spark

I don't think this function name is covered in the SQL standard either.

---

[Original implementations in dbt-utils](https://github.com/dbt-labs/dbt-utils/pull/595/files#diff-468c4d02cc4267759ef6b29498dd74f86c4744970578101b2f1d25982115f025)

In an alternate universe:
- [array_cat](https://www.youtube.com/watch?v=Rg7v-6E50IE&t=144s)

Reply:
such is life (and sql standards)!